x/incentive/keeper/store.go

```diff
@@ -0,0 +1,274 @@
+	return store.GetUint64(k.KVStore(ctx),
+		keyPrefixParamUnbondingDurationMiddle, 0, "middle unbonding duration")
+}
+
+// GetUnbondingDurationShort gets the duration in seconds of the short bonding tier.
+func (k Keeper) GetUnbondingDurationShort(ctx sdk.Context) uint64 {
+	return store.GetUint64(k.KVStore(ctx),
+		keyPrefixParamUnbondingDurationShort, 0, "short unbonding duration")
+}
+
+// GetTierWeightShort gets the ratio of rewards received by the short tier of bonded assets. Ranges 0 - 1.
+func (k Keeper) GetTierWeightMiddle(ctx sdk.Context) sdk.Dec {
+	return store.GetDec(k.KVStore(ctx),
+		keyPrefixParamTierWeightMiddle, sdk.ZeroDec(), "middle tier weight")
+}
+
+// GetTierWeightShort gets the ratio of rewards received by the middle tier of bonded assets. Ranges 0 - 1.
+func (k Keeper) GetTierWeightShort(ctx sdk.Context) sdk.Dec {
+	return store.GetDec(k.KVStore(ctx),
+		keyPrefixParamTierWeightShort, sdk.ZeroDec(), "short tier weight")
+}
+
+// GetCommunityFundAddress retrieves the community fund address parameter. It is guaranteed to be
+// either valid (by sdk.ValidateAddressFormat) or empty.
+func (k Keeper) GetCommunityFundAddress(ctx sdk.Context) sdk.AccAddress {
+	return store.GetAddress(k.KVStore(ctx),
+		keyPrefixParamCommunityFundAddress, "community fund address")
+}
+
+// setParams validates and sets the incentive module parameters
+func (k Keeper) setParams(ctx sdk.Context, params incentive.Params) error {
+	kvs := k.KVStore(ctx)
+	if err := params.Validate(); err != nil {
+		return err
+	}
+	err := store.SetUint32(kvs, keyPrefixParamMaxUnbondings,
+		params.MaxUnbondings, 0, "max unbondings")
+	if err != nil {
+		return err
+	}
+	err = store.SetUint64(kvs, keyPrefixParamUnbondingDurationLong,
+		params.UnbondingDurationLong, 0, "long unbonding duration")
+	if err != nil {
+		return err
+	}
+	err = store.SetUint64(kvs, keyPrefixParamUnbondingDurationMiddle,
+		params.UnbondingDurationMiddle, 0, "middle unbonding duration")
+	if err != nil {
+		return err
+	}
+	err = store.SetUint64(kvs, keyPrefixParamUnbondingDurationShort,
+		params.UnbondingDurationShort, 0, "short unbonding duration")
+	if err != nil {
+		return err
+	}
+	err = store.SetDec(kvs, keyPrefixParamTierWeightMiddle,
+		params.TierWeightMiddle, sdk.ZeroDec(), "middle tier weight")
+	if err != nil {
+		return err
+	}
+	err = store.SetDec(kvs, keyPrefixParamTierWeightShort,
+		params.TierWeightShort, sdk.ZeroDec(), "short tier weight")
+	if err != nil {
+		return err
+	}
+	err = store.SetAddress(kvs, keyPrefixParamCommunityFundAddress,
+		sdk.MustAccAddressFromBech32(params.CommunityFundAddress), "community fund address")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetIncentiveProgram gets an incentive program by ID, regardless of the program's status.
+// Returns the program's status if found, or an error if it does not exist.
+func (k Keeper) GetIncentiveProgram(ctx sdk.Context, id uint32) (
```

Comment:
This concept of storing incentive programs in separate lists will help iterate through them in practice - for any given query or function (e.g. EndBlocker), the module will want to iterate over upcoming, ongoing, and completed programs at different times.

The upcoming and ongoing lists in particular will benefit from not being cluttered by completed or cancelled programs.